CPE200Lab1/CPE200Lab1/CalculatorEngine.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CPE200Lab1
+{
+	class CalculatorEngine
+	{
+		private string display = "0";
+		private bool hasDot;
+		private bool isAllowBack;
+		private bool isAfterOperater;
+		private bool isAfterEqual;
+		private string firstOperand;
+		public string figure;
+		private string operate;
+
+		
+
+		public string Display()
+		{
+			return display;
+		}
+
+		public void resetAll()
+		{
+		 isAllowBack = true;
+		 hasDot = false;
+		 isAfterOperater = false;
+		 isAfterEqual = false;
+		 display = "0";
+	    }
+
+		public void point()
+		{
+			if (display is "Error")
+			{
+				return;
+			}
+			if (isAfterEqual)
+			{
+				resetAll();
+			}
+			if (display.Length is 8)
+			{
+				return;
+			}
+			if (!hasDot)
+			{
+				display += ".";
+				hasDot = true;
+			}
+		}
+
+		public void removeLastDigit()
+		{
+			if (display is "Error")
+			{
+				return;
+			}
+			if (isAfterEqual)
+			{
+				return;
+			}
+			if (!isAllowBack)
+			{
+				return;
+			}
+			if (display != "0")
+			{
+				string current = display;
+				char rightMost = current[current.Length - 1];
+				if (rightMost is '.')
+				{
+					hasDot = false;
+				}
+				display = current.Substring(0, current.Length - 1);
+				if (display is "" || display is "-")
+				{
+					display = "0";
+				}
+			}
+		}
+		public void amount()
+		{
+			if (display is "Error")
+			{
+				return;
+			}
+			if (isAfterEqual)
+			{
+				resetAll();
+			}
+			if (isAfterOperater)
+			{
+				display = "0";
+			}
+			if (display.Length is 8)
+			{
+				return;
+			}
+			isAllowBack = true;
+			if (display is "0")
+			{
+				display = "";
+			}
+			display += figure;
+			isAfterOperater = false;
+		}
+		public string calculate(string operate, string firstOperand, string secondOperand, int maxOutputSize = 8)
+		{
+			switch (operate)
+			{
+				case "+":
+					return (Convert.ToDouble(firstOperand) + Convert.ToDouble(secondOperand)).ToString();
+				case "-":
+					return (Convert.ToDouble(firstOperand) - Convert.ToDouble(secondOperand)).ToString();
+				case "X":
+					return (Convert.ToDouble(firstOperand) * Convert.ToDouble(secondOperand)).ToString();
+				case "÷":
+					// Not allow devide be zero
+					if (secondOperand != "0")
```

Comment:
What if secondOperand is 0, what will happen after user click "=" ? nothing?  it should show Error, right?